src/schemaType/EmbeddedType.ts

```diff
@@ -48,15 +47,10 @@ class EmbeddedType extends BaseSchemaType {
 	}
 
 	/** Validate the embedded document */
-	public async validate(document: Document): Promise<string[]> {
+	public async validate(document: Document): Promise<Map<string, string[]>> {
 		// - validation against the embedded document will return a single object with 0 to n keys - only those with keys indicate errors;
-		// - iterate the returned object and return the messages from each
-		// - flatten the final results
 		const documentErrors = await document.validate();
-
-		return Array.from(documentErrors.values())
-			.map((err) => ensureArray(err))
-			.flat();
+		return documentErrors;
```

Comment:
This method can be changed to no longer have async/await and just return the result